src/tools/protocol/types/ArrayType.vue

```diff
@@ -0,0 +1,73 @@
+<script setup lang="ts">
+import { inject, ref } from 'vue'
+import { I18nT, useI18n } from 'vue-i18n'
+import { getAsPrimitiveProtocol } from '../constants.ts'
+import { useGlobalState } from '../state.ts'
+import TypeChoice from './TypeChoice.vue'
+
+interface ArrayTypeDefinition {
+  countType?: string
+  count?: number | string
+  type: object | string
+}
+
+const props = defineProps<{ data: object; version: number }>()
+const { t } = useI18n()
+const state = useGlobalState()
+const scope = inject('scope') as string
+
+const errorState =
+  !Array.isArray(props.data) || props.data[0] !== 'array' || props.data.length !== 2
+const content = (props.data as any[])[1] as ArrayTypeDefinition
+const primitive = getAsPrimitiveProtocol(content.type)
+
+const showSubType = ref(false)
+</script>
+<template>
+  <div class="complex-padding flex">
+    <span v-if="errorState" class="error-state">{{ t('protocol.error.data') }}</span>
+    <I18nT
+      v-else
+      :keypath="primitive ? 'protocol.type.array.primitive' : 'protocol.type.array.complex'"
+      tag="span"
+      class="flex-1"
+    >
+      <template v-if="primitive" #type>{{ t(`protocol.type.${primitive}`) }}</template>
+      <template #count>
+        <I18nT
+          :keypath="`protocol.type.array.count_${content.countType ? 'type' : 'fixed'}`"
+          tag="span"
+        >
+          <template v-if="content.countType" #type>
+            {{ t(`protocol.type.${content.countType}`) }}
+          </template>
+          <template v-else #count>
+            <span
+              v-if="typeof content.count === 'string'"
+              class="underline italic cursor-default"
+              @mouseover="state.selectName(content.count, scope)"
+              @mouseout="state.unselectName(content.count, scope)"
+            >
+              {{ content.count }}
+            </span>
+            <span v-else>{{ content.count }}</span>
+          </template>
+        </I18nT>
+      </template>
+    </I18nT>
+    <span
```

Comment:
It is not immediately obvious that the [expand] and [Search ID] buttons can be pressed. Consider applying a link color to it.